hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -157,22 +158,22 @@ public ContainerDiffReport diff(KeyValueContainerData thisContainer,
         Preconditions.assertNotNull(peerChecksumInfo, "Peer checksum info is null");
         Optional<ContainerProtos.ContainerChecksumInfo> thisContainerChecksumInfo = read(thisContainer);
         if (!thisContainerChecksumInfo.isPresent()) {
-          throw new SCMException("The container #" + thisContainer.getContainerID() +
-              " doesn't have container checksum", SCMException.ResultCodes.IO_EXCEPTION);
+          throw new StorageContainerException("The container #" + thisContainer.getContainerID() +
+              " doesn't have container checksum", ContainerProtos.Result.IO_EXCEPTION);
```

Comment:
We are using the same result code for this case and the legitimate IO exception case. This might be okay because we could want the caller to queue the scanner for on demand scanning in both cases. We can leave it for now and revisit in the follow up patch that needs to handle this if required.